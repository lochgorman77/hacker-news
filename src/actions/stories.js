import { HackerNewsApi } from 'api/config';
import { storiesActions, detailsActions } from './../constants/ActionTypes';

export function fetchTopStoriesSuccess(stories) {
  return {
    type: storiesActions.FETCH_TOP_STORIES_SUCCESS,
    payload: stories,
  };
}
export function fetchTopStoriesError(error) {
  return {
    type: storiesActions.FETCH_TOP_STORIES_ERROR,
    payload: error,
  };
}
export function fetchTopStories() {
  return function (dispatch) {
    dispatch({ type: storiesActions.FETCH_TOP_STORIES });
    dispatch({ type: detailsActions.RESET });

    return HackerNewsApi.get(`/topstories.json`)
      .then((result) => {
        dispatch(fetchTopStoriesSuccess(result.data));
      })
      .catch((error) => {
        dispatch(fetchTopStoriesError(error));
      });
  };
}

export function fetchNewStoriesSuccess(stories) {
  return {
    type: storiesActions.FETCH_NEW_STORIES_SUCCESS,
    payload: stories,
  };
}
export function fetchNewStoriesError(error) {
  return {
    type: storiesActions.FETCH_NEW_STORIES_ERROR,
    payload: error,
  };
}
export function fetchNewStories() {
  return function (dispatch) {
    dispatch({ type: storiesActions.FETCH_NEW_STORIES });
    dispatch({ type: detailsActions.RESET });

    return HackerNewsApi.get(`/newstories.json`)
      .then((result) => {
        dispatch(fetchNewStoriesSuccess(result.data));
      })
      .catch((error) => {
        dispatch(fetchNewStoriesError(error));
      });
  };
}
