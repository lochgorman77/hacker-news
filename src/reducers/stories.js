import { storiesActions, pageSize } from 'constants/ActionTypes';

const initialState = {
  top: [],
  new: [],
};

export default function roles(state = initialState, action) {
  switch (action.type) {
    case storiesActions.FETCH_TOP_STORIES_SUCCESS:
      return {
        ...state,
        top: action.payload,
      };
    case storiesActions.FETCH_TOP_STORIES_ERROR:
      return {
        ...state,
        top: initialState.top,
      };
    case storiesActions.FETCH_NEW_STORIES_SUCCESS:
      return {
        ...state,
        new: action.payload,
      };
    case storiesActions.FETCH_NEW_STORIES_ERROR:
      return {
        ...state,
        new: initialState.new,
      };
    default:
      return state;
  }
}
export const getTopStories = (state) => state.stories.top;
export const getNewStories = (state) => state.stories.new;
export const getPageCount = (state) => {
  return 500 / pageSize;
};
