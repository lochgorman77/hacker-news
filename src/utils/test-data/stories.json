[
    25429627,
    25429623,
    25429620,
    25429614,
    25429530,
    25429518,
    25429516,
    25429515,
    25429513,
    25429501,
    25429493,
    25429472,
    25429469,
    25429466,
    25429433,
    25429431,
    25429416,
    25429411,
    25429407,
    25429406,
    25429394,
    25429385,
    25429376,
    25429375,
    25429373,
    25429363,
    25429337,
    25429320,
    25429313,
    25429298,
    25429277,
    25429271,
    25429255,
    25429244,
    25429243,
    25429241,
    25429237,
    25429228,
    25429220,
    25429218,
    25429208,
    25429199,
    25429195,
    25429193,
    25429163,
    25429162,
    25429159,
    25429153,
    25429148,
    25429146,
    25429142,
    25429139,
    25429134,
    25429133,
    25429132,
    25429119,
    25429112,
    25429096,
    25429088,
    25429080,
    25429077,
    25429062,
    25429044,
    25429036,
    25428997,
    25428996,
    25428990,
    25428960,
    25428959,
    25428951,
    25428918,
    25428912,
    25428900,
    25428894,
    25428881,
    25428866,
    25428862,
    25428842,
    25428829,
    25428802,
    25428800,
    25428775,
    25428769,
    25428763,
    25428750,
    25428748,
    25428738,
    25428733,
    25428706,
    25428703,
    25428695,
    25428676,
    25428669,
    25428653,
    25428648,
    25428629,
    25428621,
    25428620,
    25428618,
    25428602,
    25428591,
    25428583,
    25428579,
    25428572,
    25428570,
    25428566,
    25428562,
    25428559,
    25428550,
    25428544,
    25428543,
    25428536,
    25428532,
    25428527,
    25428515,
    25428509,
    25428496,
    25428495,
    25428485,
    25428484,
    25428481,
    25428474,
    25428469,
    25428464,
    25428454,
    25428453,
    25428429,
    25428417,
    25428413,
    25428397,
    25428390,
    25428389,
    25428369,
    25428362,
    25428345,
    25428339,
    25428309,
    25428302,
    25428298,
    25428285,
    25428275,
    25428268,
    25428263,
    25428260,
    25428252,
    25428249,
    25428215,
    25428200,
    25428194,
    25428177,
    25428173,
    25428170,
    25428161,
    25428153,
    25428112,
    25428110,
    25428091,
    25428071,
    25428066,
    25428063,
    25428046,
    25428036,
    25428020,
    25428016,
    25428011,
    25427996,
    25427993,
    25427982,
    25427964,
    25427956,
    25427952,
    25427950,
    25427946,
    25427940,
    25427932,
    25427925,
    25427917,
    25427898,
    25427894,
    25427875,
    25427863,
    25427843,
    25427830,
    25427829,
    25427819,
    25427813,
    25427812,
    25427811,
    25427798,
    25427790,
    25427783,
    25427779,
    25427756,
    25427750,
    25427745,
    25427701,
    25427638,
    25427633,
    25427631,
    25427598,
    25427585,
    25427577,
    25427574,
    25427552,
    25427536,
    25427534,
    25427520,
    25427507,
    25427504,
    25427489,
    25427486,
    25427483,
    25427478,
    25427471,
    25427467,
    25427418,
    25427414,
    25427413,
    25427398,
    25427396,
    25427381,
    25427365,
    25427338,
    25427316,
    25427314,
    25427286,
    25427285,
    25427281,
    25427276,
    25427265,
    25427247,
    25427238,
    25427236,
    25427231,
    25427218,
    25427212,
    25427210,
    25427208,
    25427204,
    25427196,
    25427194,
    25427183,
    25427169,
    25427163,
    25427159,
    25427110,
    25427100,
    25427089,
    25427086,
    25427084,
    25427068,
    25427067,
    25427060,
    25427055,
    25427038,
    25427015,
    25427013,
    25427000,
    25426998,
    25426977,
    25426976,
    25426924,
    25426916,
    25426898,
    25426831,
    25426780,
    25426752,
    25426747,
    25426739,
    25426714,
    25426712,
    25426694,
    25426684,
    25426662,
    25426617,
    25426613,
    25426610,
    25426607,
    25426606,
    25426593,
    25426582,
    25426579,
    25426566,
    25426565,
    25426559,
    25426553,
    25426547,
    25426534,
    25426519,
    25426514,
    25426511,
    25426507,
    25426498,
    25426494,
    25426493,
    25426472,
    25426471,
    25426463,
    25426459,
    25426444,
    25426439,
    25426432,
    25426421,
    25426401,
    25426397,
    25426353,
    25426323,
    25426308,
    25426300,
    25426285,
    25426275,
    25426271,
    25426232,
    25426226,
    25426217,
    25426207,
    25426198,
    25426195,
    25426185,
    25426180,
    25426174,
    25426161,
    25426155,
    25426140,
    25426128,
    25426098,
    25426085,
    25426078,
    25426074,
    25426066,
    25426055,
    25426044,
    25426037,
    25426035,
    25426034,
    25426030,
    25426004,
    25426002,
    25425990,
    25425939,
    25425895,
    25425884,
    25425874,
    25425867,
    25425864,
    25425844,
    25425838,
    25425830,
    25425829,
    25425809,
    25425804,
    25425794,
    25425783,
    25425780,
    25425769,
    25425721,
    25425718,
    25425699,
    25425652,
    25425647,
    25425646,
    25425634,
    25425628,
    25425579,
    25425547,
    25425545,
    25425529,
    25425524,
    25425511,
    25425491,
    25425487,
    25425484,
    25425469,
    25425462,
    25425442,
    25425421,
    25425420,
    25425412,
    25425408,
    25425403,
    25425402,
    25425389,
    25425372,
    25425371,
    25425361,
    25425327,
    25425312,
    25425311,
    25425306,
    25425293,
    25425291,
    25425289,
    25425282,
    25425280,
    25425255,
    25425237,
    25425234,
    25425199,
    25425196,
    25425181,
    25425167,
    25425151,
    25425148,
    25425116,
    25425074,
    25425058,
    25425020,
    25425016,
    25425006,
    25425004,
    25424993,
    25424992,
    25424990,
    25424987,
    25424967,
    25424966,
    25424965,
    25424923,
    25424907,
    25424895,
    25424852,
    25424767,
    25424755,
    25424738,
    25424735,
    25424722,
    25424709,
    25424703,
    25424683,
    25424682,
    25424672,
    25424667,
    25424662,
    25424632,
    25424619,
    25424618,
    25424614,
    25424611,
    25424595,
    25424579,
    25424562,
    25424560,
    25424541,
    25424534,
    25424521,
    25424519,
    25424518,
    25424515,
    25424507,
    25424505,
    25424499,
    25424489,
    25424478,
    25424468,
    25424438,
    25424434,
    25424430,
    25424419,
    25424414,
    25424361,
    25424358,
    25424355,
    25424344,
    25424277,
    25424268,
    25424250,
    25424246,
    25424225,
    25424221,
    25424209,
    25424200,
    25424180,
    25424171,
    25424164,
    25424149,
    25424145,
    25424134,
    25424126,
    25424116,
    25424115,
    25424111,
    25424103,
    25424088,
    25424065,
    25424053,
    25424034,
    25424022,
    25424015,
    25424009,
    25424004,
    25423979,
    25423964,
    25423938,
    25423931,
    25423926,
    25423917,
    25423912,
    25423901,
    25423892,
    25423891
]
