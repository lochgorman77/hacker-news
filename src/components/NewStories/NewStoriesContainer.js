import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Details from 'components/Details/Details';

import { fetchNewStories } from 'actions/stories';
import { getNewStories } from 'reducers/stories';
import { fetchDetails } from 'actions/details';
import { getDetails } from 'reducers/details';
import { pageSize } from 'constants/ActionTypes';

export class NewStoriesContainer extends Component {
  componentDidMount() {
    this.props.handleFetchNewStories();
  }

  componentDidUpdate(prevProps) {
    if (this.props.stories !== prevProps.stories) {
      for (let i = 0; i < pageSize; i++) {
        this.props.handleFetchDetails(this.props.stories[i]);
      }
    }
  }

  render() {
    return <Details details={this.props.details}></Details>;
  }
}

NewStoriesContainer.propTypes = {
  handleFetchNewStories: PropTypes.func,
  handleFetchDetails: PropTypes.func,
  stories: PropTypes.array,
  details: PropTypes.array,
};

const mapStateToProps = (state) => ({
  stories: getNewStories(state),
  details: getDetails(state),
});

const mapDispatchToProps = (dispatch) => ({
  handleFetchNewStories: () => dispatch(fetchNewStories()),
  handleFetchDetails: (id) => dispatch(fetchDetails(id)),
});

const ConnectedNewStoriesContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(NewStoriesContainer);

export default ConnectedNewStoriesContainer;
