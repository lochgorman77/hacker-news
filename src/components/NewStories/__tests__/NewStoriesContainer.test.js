import React from 'react';
import { NewStoriesContainer } from '../NewStoriesContainer';
import stories from 'utils/test-data/stories-new.json';

const props = {
  handleFetchNewStories: jest.fn(),
  handleFetchDetails: jest.fn(),
  stories: [],
  details: [],
};

describe('<NewStoriesContainer />', () => {
  let shallow, wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<NewStoriesContainer {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe('Details');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('component updated ', () => {
    wrapper.setProps({ stories: stories });
    expect(wrapper.instance().props.handleFetchDetails).toBeCalled();
  });
});
