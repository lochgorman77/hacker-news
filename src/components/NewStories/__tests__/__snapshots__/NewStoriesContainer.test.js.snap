// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewStoriesContainer /> renders correctly 1`] = `
<Details
  details={Array []}
/>
`;
